### PR TITLE
Update system-metrics.rst

### PR DESCRIPTION
**Pull request by anyscalesam**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-observability/reference/system-metrics.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ Ray exports a number of system metrics, which provide introspection into the sta
...
    * - `ray_object_store_memory`
      - `Location`, `ObjectState`, `InstanceId`
-     - Object store memory usage in bytes, `broken down <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L231>`_ by logical Location (SPILLED, IN_MEMORY, etc.), and ObjectState (UNSEALED, SEALED).
+     - Object store memory usage in bytes, `broken down <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L231>`_ by logical Location (SPILLED, MMAP_DISK, MMAP_SHM, and WORKER_HEAP). Definitions are as follows: SPILLED = Objects that have spilled to disk or a remote Storage solution (eg: AWS S3); MMAP_DISK = Objects stored on a memory-mapped page on disk; MMAP_SHM: Objects store on a memory-mapped page in Shared Memory; WORKER_HEAP: Objects, usually smaller, stored in the memory of the Ray Worker process itself.
```
Comment (rkooo567) [inline doc/source/ray-observability/reference/system-metrics.rst:28]:
```suggestion
     - Object store memory usage in bytes, `broken down <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L231>`_ by logical Location (SPILLED, MMAP_DISK, MMAP_SHM, and WORKER_HEAP). Definitions are as follows: SPILLED = Objects that have spilled to disk or a remote Storage solution (eg: AWS S3). The default is the disk; MMAP_DISK = Objects stored on a memory-mapped page on disk. It is very slow and only happens under severe memory pressure; MMAP_SHM: Objects store on a memory-mapped page in Shared Memory. It is the default mode when there's no memory pressure; WORKER_HEAP: Objects, usually smaller, stored in the memory of the Ray Worker process itself. Small objects are stored in worker heap. 
```

**Review by rkooo567 — APPROVED** [review]
Actually, I feel like we need a holistic doc that explains this better and add a link here.

**Comment by anyscalesam** [comment]
@rkooo567 can you merge once it's been approved; i don't have rights

**Comment by angelinalg** [comment]
@rkooo567 what do you think of the formatting and copy edit suggestions I made? If you want to ignore, I can just merge. LMK.

File: doc/source/ray-observability/reference/system-metrics.rst [hunk 2]
```diff
@@ -25,7 +25,11 @@ Ray exports a number of system metrics, which provide introspection into the sta
      - Logical resource usage for each node of the cluster. Each resource has some quantity that is `in either <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L188>`_ USED state vs AVAILABLE state. The Name label defines the resource name (e.g., CPU, GPU).
    * - `ray_object_store_memory`
      - `Location`, `ObjectState`, `InstanceId`
-     - Object store memory usage in bytes, `broken down <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L231>`_ by logical Location (SPILLED, IN_MEMORY, etc.), and ObjectState (UNSEALED, SEALED).
+     - Object store memory usage in bytes, `broken down <https://github.com/ray-project/ray/blob/9eab65ed77bdd9907989ecc3e241045954a09cb4/src/ray/stats/metric_defs.cc#L231>`_ by logical Location (SPILLED, MMAP_DISK, MMAP_SHM, and WORKER_HEAP). Definitions are as follows: 
+         - SPILLED - Objects that have spilled to disk or a remote Storage solution (for example, AWS S3). The default is the disk.
```
Comment (angelinalg) [inline doc/source/ray-observability/reference/system-metrics.rst:29]:
```suggestion
SPILLED - Objects that have spilled to disk or a remote Storage solution (for example, AWS S3). The default is the disk.
```

File: doc/source/ray-observability/reference/system-metrics.rst [hunk 3]
```diff
@@ -25,7 +25,11 @@ Ray exports a number of system metrics, which provide introspection into the sta
...
+         - MMAP_DISK - Objects stored on a memory-mapped page on disk. This mode very slow and only happens under severe memory pressure.
```
Comment (angelinalg) [inline doc/source/ray-observability/reference/system-metrics.rst:30]:
```suggestion
         MMAP_DISK - Objects stored on a memory-mapped page on disk. This mode very slow and only happens under severe memory pressure.
```

File: doc/source/ray-observability/reference/system-metrics.rst [hunk 4]
```diff
@@ -25,7 +25,11 @@ Ray exports a number of system metrics, which provide introspection into the sta
...
+         - MMAP_SHM - Objects store on a memory-mapped page in Shared Memory. This mode is the default, in the absence of memory pressure.
```
Comment (angelinalg) [inline doc/source/ray-observability/reference/system-metrics.rst:31]:
```suggestion
         MMAP_SHM - Objects store on a memory-mapped page in Shared Memory. This mode is the default, in the absence of memory pressure.
```

File: doc/source/ray-observability/reference/system-metrics.rst [hunk 5]
```diff
@@ -25,7 +25,11 @@ Ray exports a number of system metrics, which provide introspection into the sta
...
+         - WORKER_HEAP - Objects, usually smaller, stored in the memory of the Ray Worker process itself. Small objects are stored in the worker heap.
```
Comment (angelinalg) [inline doc/source/ray-observability/reference/system-metrics.rst:32]:
```suggestion
         WORKER_HEAP - Objects, usually smaller, stored in the memory of the Ray Worker process itself. Small objects are stored in the worker heap. 
```

**Review by angelinalg — APPROVED** [review]
Trying to fix formatting problem.

**Review by angelinalg — APPROVED** [review]

**Pull request merged**